src/object.js

```diff
@@ -146,11 +156,31 @@ ReflectionObject.prototype.onRemove = function onRemove(parent) {
 ReflectionObject.prototype.resolve = function resolve() {
     if (this.resolved)
         return this;
-    if (this.root instanceof Root)
-        this.resolved = true; // only if part of a root
+    this._resolveFeatures();
+    if (this.root instanceof Root || this.parent)
```

Comment:
Is this sufficient?

- We probably don't want to call `_resolveFeatures()` unless this condition is true
- What if the parent is itself orphaned?

Reply:
- Moved _resolveFeatures, I think you're right.
- If the parent itself is orphaned, then this wouldn't get called recursively since this usually only gets called under `resolveAll`.